source/lac/petsc_precondition.cc

```diff
@@ -57,23 +53,16 @@ namespace PETScWrappers
       {}
   }
 
-
   void
   PreconditionBase::clear()
   {
-    mpi_communicator = MPI_COMM_NULL;
-
-    matrix = nullptr;
-
-    if (pc != nullptr)
+    if (pc)
       {
         PetscErrorCode ierr = PCDestroy(&pc);
-        pc                  = nullptr;
```

Comment:
Why remove the assignment?

Reply:
because `PCDestroy` already does it for you (all `XXXDestroy` methods in PETSc nullify the pointer to avoid reusing the freed memory)